jupyter_server/services/sessions/sessionmanager.py

```diff
@@ -13,13 +13,19 @@
 
 from traitlets.config.configurable import LoggingConfigurable
 from traitlets import Instance
+from traitlets import Unicode
 
 from jupyter_server.utils import ensure_async
 from jupyter_server.traittypes import InstanceFromClasses
 
 
 class SessionManager(LoggingConfigurable):
 
+    database_path = Unicode(
```

Comment:
Since this doesn't work if pointing at a directory, I suggest it be named `database_filepath` and that we perhaps add a valiator to ensure the value is not a directory.  Perhaps adding an example to the help string like "/path/to/my/dbfile.db" would also clarify the expected value.

It might be good to reference the meaning of ":memory:" in the help string as well.

FWIW, there's the log entry if given a directory:
```
[E 2021-11-19 14:44:09.911 ServerApp] Uncaught exception GET /api/sessions?1637361849903 (::1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/api/sessions?1637361849903', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/tornado/web.py", line 1704, in _execute
        result = await result
      File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/jupyter_server/services/sessions/handlers.py", line 26, in get
        sessions = await ensure_async(sm.list_sessions())
      File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/jupyter_server/utils.py", line 189, in ensure_async
        result = await obj
      File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/jupyter_server/services/sessions/sessionmanager.py", line 266, in list_sessions
        c = self.cursor.execute("SELECT * FROM session")
      File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/jupyter_server/services/sessions/sessionmanager.py", line 46, in cursor
        self._cursor = self.connection.cursor()
      File "/opt/anaconda3/envs/elyra-dev/lib/python3.8/site-packages/jupyter_server/services/sessions/sessionmanager.py", line 58, in connection
        self._connection = sqlite3.connect(self.database_path, isolation_level=None)
    sqlite3.OperationalError: unable to open database file
```

Reply:
I agree, I think validating that the path isn't a directory would be helpful. 
If we're expecting the database to already exist before calling `connect`, I also think it'll be helpful to check if the file is a sqlite db because `sqlite3` will automatically create a new database during `connect()` if it doesn't exist.

Reply:
Agree, just add a valid function or provide a default filename if path is a directory would be helpful.

Reply:
Thanks, all!

Renaming to `database_filepath` is a good idea. This should be more specific. I'll also add some documentation describing the meaning of `:memory:`

I'll work on adding validation per @mwakaba2's and @Wh1isper's comments too.